File: quiz_app_flutter/lib/presentation/quiz/intent/quiz_intent.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:quiz_app_flutter/presentation/quiz/route/quiz_screen_data.dart';

part 'quiz_intent.freezed.dart';

@freezed
class QuizIntent with _$QuizIntent {
  const factory QuizIntent.loadQuiz(QuizScreenData data) = LoadQuiz;

  const factory QuizIntent.nextQuestion() = NextQuestion;

  const factory QuizIntent.skipQuestion() = SkipQuestion;

  const factory QuizIntent.updateSelectedAnswers(List<int> answers) = UpdateSelectedAnswers;

  const factory QuizIntent.submitAnswer() = SubmitAnswer;
}

File: quiz_app_kmp/composeApp/src/commonMain/kotlin/com/tnm/quizappkmp/presentation/quiz/intent/QuizIntent.kt
package com.tnm.quizappkmp.presentation.quiz.intent

import com.tnm.quizappkmp.presentation.quiz.route.QuizScreenData

sealed class QuizIntent {
    data class LoadQuiz(val data: QuizScreenData) : QuizIntent()
    object SubmitAnswer : QuizIntent()
    object SkipQuestion : QuizIntent()
    object NextQuestion : QuizIntent()
    data class UpdateSelectedAnswers(val answers: List<Int>) : QuizIntent()
    object NavigateToResult : QuizIntent()
}

sealed class QuizNavEvent {
    data class NavigateToResult(val key: String) : QuizNavEvent()
}